File: apps/web/src/types/todo.ts
export type Todo = {
  id: number;
  title: string;
  description: string;
  completed: boolean;
  createdAt: string;
  updatedAt: string;
};

export type TodoPayload = {
  title: string;
  description: string;
};

export type TodoPatchPayload = Partial<Pick<Todo, "title" | "description" | "completed">>;

File: apps/api/internal/domain/todo.go
package domain

import (
	"time"
)

type Todo struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Completed   bool      `json:"completed"`
	CreatedAt   time.Time `json:"createdAt,format:'2006-01-02T15:04:05Z'"`
	UpdatedAt   time.Time `json:"updatedAt,format:'2006-01-02T15:04:05Z'"`
}
